token/core/zkatdlog/nogh/tokens.go

```diff
@@ -128,3 +140,79 @@ func supportedTokenFormat(pp *crypto.PublicParams) (token.Format, error) {
 	}
 	return token.Format(hasher.HexDigest()), nil
 }
+
+func (s *TokensService) ProcessTokenConversionRequest(utp *driver.TokenConversionRequest) ([]token.Type, []uint64, error) {
+	if utp == nil {
+		return nil, nil, errors.New("nil token conversion request")
+	}
+
+	ok, err := s.CheckConversionProof(utp.Proof, utp.Proof, utp.Tokens)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to check conversion proof")
+	}
+	if !ok {
+		return nil, nil, errors.New("invalid conversion proof")
+	}
+
+	var tokenTypes []token.Type
+	var tokenValue []uint64
+
+	// which types do we recognize? Any other type is not convertable
+	fabtoken16Type, err := fabtoken2.SupportedTokenFormat(16)
```

Comment:
These can also be defined as constants outside of this method
```
const fabtoken16Type = utils.MustGet(fabtoken2.SupportedTokenFormat(16))
```

Reply:
Is this supposed to be fabtoken? I think the format for zkat was comm.

Reply:
the dlog driver supports the conversion from fabtoken, it might support more. To arrive later.